enterprise_access/apps/subsidy_access_policy/utils.py

```diff
@@ -140,3 +148,56 @@ def cents_to_usd_string(cents):
     if cents is None:
         return None
     return "${:,.2f}".format(float(cents) / constants.CENTS_PER_DOLLAR)
+
+
+def validate_retired_budget_deactivation(policy_instance):
+    """
+    Raise a ValidationError if a policy would become both retired and inactive while having existing spend.
+    The error is attached to the field(s) being changed: 'active', 'retired', or both.
+    Uses original values from the database to detect changes.
+    """
+    if policy_instance.pk:
+        try:
+            original = type(policy_instance).objects.get(pk=policy_instance.pk)
+            original_active = original.active
+            original_retired = original.retired
+        except ObjectDoesNotExist:
+            original_active = policy_instance.active
+            original_retired = policy_instance.retired
+    else:
+        original_active = policy_instance.active
+        original_retired = policy_instance.retired
+
+    active_will_change = policy_instance.active != original_active
+    retired_will_change = policy_instance.retired != original_retired
+
+    will_be_retired_and_inactive = (
+        (original_retired and not policy_instance.active and active_will_change and not retired_will_change) or
+        (policy_instance.retired and not policy_instance.active and retired_will_change)
+    )
+    if not will_be_retired_and_inactive:
+        return
```

Comment:
nit: possibly this will be clearer by adding more variables, something like
```python
would_make_retired_policy_inactive = (
    original_retired and not policy_instance.active and active_will_change and not retired_will_change
)
would_make_inactive_policy_retired = (
    policy_instance.retired and not policy_instance.active and retired_will_change
)
if not (would_make_retired_policy_inactive or would_make_inactive_policy_retired):
    return
```

Reply:
Updated!